enhancements/sig-architecture/149-moving-cluster-proxy-to-core/README.md

```diff
@@ -0,0 +1,167 @@
+capabilities into the klusterlet.
+
+### Goals
+
+- Add a feature flag `ClusterProxy` in both klusterlet and clustermanager. When enabled, a gRPC server will
+  start on the hub, and the klusterlet agent will connect to the gRPC server for proxy functionality.
+- When the feature is enabled, ClusterManager starts an HTTP server to proxy requests to the target cluster
+  via the klusterlet agent.
+- All existing capabilities of `cluster-proxy` will be preserved.
+
+### Non-Goals
+
+- How to deprecate the `cluster-proxy addon` is not in the scope of this proposal, given that `cluster-proxy`
+  is still consumed by `MulticlusterGateway` and other consumers.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+Users can enable the `ClusterProxy` feature gate in both clustermanager and klusterlet. Once enabled, users can
+use the HTTP server on the hub cluster to proxy requests to the API server of the targeted cluster.
+
+#### Story 2
+
+Users can authenticate via proxy using either the token for the targeted cluster, or the token for the hub
+cluster if impersonation permissions are granted to the klusterlet agent.
+
+
+### Risks and Mitigation
+
+N/A
+
+## Design Details
+
+#### Registration
+The connection between klusterlet and hub for proxy functionality will remain gRPC-based. This is independent
+of the gRPC registration driver, requiring a separate gRPC configuration file for the proxy. To obtain the gRPC
+configuration for proxy, the klusterlet agent will send a Certificate Signing Request (CSR) to the hub with the
+signer name `open-cluster-management.io/klusterlet-proxy`. The gRPC server on the hub will approve and sign the
+CSR. The gRPC configuration will be saved in the `hub-kubeconfig-secret` with the key `proxy-grpc.yaml`. The
+klusterlet will only start the proxy when it detects the existence of this configuration file. The proxy
+registration process begins after the cluster registration process and will not impact normal cluster
+registration.
+
+#### Proxy API
+
+The hub cluster will start an externally accessible HTTP endpoint to proxy requests to the API server of each managed
+cluster. The API path follows the format: `https://<server address>:<server port>/<cluster name>`.
+
+Consumers can configure their kubeconfig using this API endpoint and the related CA bundle to access the API server
+of the targeted cluster.
+
+`clusteradm` will be enhanced with a subcommand `clusteradm proxy --cluster <cluster name>`, which will configure
+the kubeconfig with the proxy endpoint in a cluster-specific context and switch the kubeconfig context to the target
+cluster. Users can then use kubectl to connect to the target cluster.
+
+#### Integration with ClusterProfile API
+
+#### Installation
+
+To enable the feature, the user needs to enable the feature gate in `ClusterManager` and `Klusterlet` API.
+In addition, A new proxyConfig field should be added in `ClusterManager` API:
```

Comment:
how about introduce a config for grpcserver instead of proxyConifg
```go
type GRPCServerConfiguration struct {
    ImagePullSpec string `json:"imagePullSpec,omitempty"`
    FeatureGates []FeatureGate `json:"featureGates,omitempty"`
    EndpointExposure *GRPCEndpointExposure `json:"endpointExposure,omitempty"` 
}
```

Reply:
yes, for registrationConfig, we need  to set  authType with grpc to enable the grpc

Reply:
updated